src/test/frontend/common/components/label/label_controller_test.js

```diff
@@ -0,0 +1,58 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import componentsModule from 'common/components/components_module';
+import LabelController from 'common/components/label/label_controller';
+
+describe('Label directive controller', () => {
+  /**
+   * @type {!LabelController}
+   */
+  let ctrl;
+
+  /**
+   *  @type {!Object<string, string>}
+   */
+  let labelsMock = {
+    app: 'app',
+    version: 'version',
+    testLabel: 'test',
+  };
+
+  /**
+   * @type {!Array<Object.<string,string>>}
+   */
+  let expectedLabels = [
+    {key: 'app', value: 'app'},
+    {key: 'version', value: 'version'},
+    {key: 'testLabel', value: 'test'},
+  ];
+
+  beforeEach(() => {
+    angular.mock.module(componentsModule.name);
+
+    angular.mock.inject(($controller, $rootScope) => {
+      // This is needed to mock variables initialized from scope with bindToController property.
+      let scope = $rootScope.$new();
+      let scopeData = {
+        labels: labelsMock,
+      };
+
+      ctrl = $controller(LabelController, {$scope: scope}, scopeData);
```

Comment:
When you remove the logic from the controller, you can still have this test to verify that DOM tree has keys and values. There is at least one test in the codebase that does something similar. Look for `$compile`.